plans/sep6.md

```diff
@@ -0,0 +1,213 @@
+## SEP-6 withdraw and deposit
+
+Deposit and withdraw share a good number of steps and Transferresponse shapes,
+meaning most of the API can be consistent across both.
+
+### Types
+
+```ts
+type startWithdrawal = (transferServer: string) => PendingWithdrawal;
+type startDeposit = (transferServer: string) => PendingDeposit;
+
+type kycPrompt = (response: InteractiveKycNeeded) => Promise<KycPromptStatus>;
+type getKycUrl = (
+  response: InteractiveKycNeeded,
+  callbackUrl: string,
+) => string;
+
+class PendingTransfer {
+  supportedAssets: () => WithdrawInfo | DepositInfo;
+  getFinalFee: (args: FeeArgs) => Promise<number>;
+}
+
+class PendingWithdrawal extends PendingTransfer {
+  withdraw: (args: WithdrawRequest) => Promise<Request>;
+}
+
+class PendingDeposit extends PendingTransfer {
+  deposit: (args: DepositRequest) => Promise<Request>;
+}
+interface FeeArgs {
+  assetCode: string;
+  amount: string;
+  operation: "withdraw" | "deposit";
+  type: "string";
+}
+
+interface WithdrawAssetInfo {
+  assetCode: string;
+  fee: Fee;
+  minAmount: number;
+  maxAmount: number;
+  types: {
+    name: string;
+    fields: Field[];
+  }[];
+}
+
+interface WithdrawInfo {
+  [assetCode: string]: WithdrawAssetInfo;
+}
+
+interface DepositAssetInfo {
+  assetCode: string;
+  fee: Fee;
+  minAmount: number;
+  maxAmount: number;
+  fields: Field[];
+}
+
+interface DepositInfo {
+  [assetCode: string]: DepositAssetInfo;
+}
+
+interface WithdrawRequest {
+  type: string;
+  assetCode: string;
+  dest: string;
+  destExtra?: string;
+  account?: string;
+  memo: Memo;
+}
+
+interface DepositRequest {
+  assetCode: string;
+  account: string;
+  memo?: Memo;
+  emailAddress?: string;
+  type?: string;
+}
+
+type responseTypes =
+  | "ok"
+  | "non_interactive_customer_info_needed"
+  | "interactive_customer_info_needed"
+  | "customer_info_status"
+  | "error";
```

Comment:
Impl detail but can these use the values from stellar-sdk instead of string literals?

Reply:
At first I had typed "yes" but since this is a SEP, these strings don't seem to appear in the stellar-sdk codebase. We could add them for sure.